File: TP1 - Dart/domain/entities/clinic.dart
import './entities.dart';

class Clinic {
  final String _id;
  List<Orthodontist> _orthodontists;
  List<Assistent> _assistents;
  List<Patient> _patients;
  List<Expense> _expenses;
  Receptionist _receptionist;
  Admin _admin;

  Clinic(
    this._id,
    this._receptionist,
  )   : _orthodontists = [],
        _assistents = [],
        _patients = [],
        _expenses = [],
        _admin = Admin(_id);

  // Getter's
  String get id => this._id;
  List<Orthodontist> get orthodontists => this._orthodontists;
  List<Assistent> get assistents => this._assistents;
  List<Patient> get patients => this._patients;
  List<Expense> get expenses => this._expenses;
  Receptionist get receptionist => this._receptionist;
  Admin get admin => this._admin;

  // Setter's
  set patients(List<Patient> patients) => this._patients = patients;
  set expenses(List<Expense> expenses) => this._expenses = expenses;
  set orthodontists(List<Orthodontist> orthodontists) =>
      this._orthodontists = orthodontists;
  set assistents(List<Assistent> assistents) => this._assistents = assistents;
  set receptionist(Receptionist receptionist) =>
      this._receptionist = receptionist;
}

File: TP2 - Java/src/domain/entities/Clinic.java
package domain.entities;

import java.util.List;

public class Clinic {
  private final String id;
  private List<Orthodontist> orthodontists;
  private List<Assistent> assistents;
  private List<Patient> patients;
  private List<Expense> expenses;
  private Receptionist receptionist;
  private User loggedUser;
  private Admin admin;

  public Clinic(String id, Receptionist receptionist, Admin admin) {
    this.id = id;
    this.receptionist = receptionist;
    this.admin = admin;
  }

  public String getId() {
    return id;
  }

  public List<Orthodontist> getOrthodontists() {
    return orthodontists;
  }

  public void setOrthodontists(List<Orthodontist> orthodontists) {
    this.orthodontists = orthodontists;
  }

  public List<Assistent> getAssistents() {
    return assistents;
  }

  public void setAssistents(List<Assistent> assistents) {
    this.assistents = assistents;
  }

  public List<Patient> getPatients() {
    return patients;
  }

  public void setPatients(List<Patient> patients) {
    this.patients = patients;
  }

  public List<Expense> getExpenses() {
    return expenses;
  }

  public void setExpenses(List<Expense> expenses) {
    this.expenses = expenses;
  }

  public Receptionist getReceptionist() {
    return receptionist;
  }

  public void setReceptionist(Receptionist receptionist) {
    this.receptionist = receptionist;
  }

  public User getLoggedUser() {
    return loggedUser;
  }

  public void setLoggedUser(User loggedUser) {
    this.loggedUser = loggedUser;
  }

  public Admin getAdmin() {
    return admin;
  }

  public void setAdmin(Admin admin) {
    this.admin = admin;
  }
}
